WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -1866,38 +2261,51 @@ static GSourceFuncs _handlerIntervention =
             }
         }
 
+        void DeactivateBrowser(PluginHost::IShell::reason reason) {
```

Comment:
This should kill a particular thread, glib event loop thread.

Reply:
It does kill, but now in a generic and controlled way.